### PR TITLE
Use doxygen-awesome template for the documentation

### PR DESCRIPTION
**Pull request by pcafrica**
As the title says (https://jothepro.github.io/doxygen-awesome-css/).

The primary color has been set equal to the one in the titlebar of https://new.dealii.org.

A couple of screenshots:

Screenshot from 2024-08-16 14-31-14
Screenshot from 2024-08-16 14-33-22

Of course many more customizations are available.

@tamiko @tjhei

### CODE REVIEW

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
awesome, thank you!

**Review by tamiko — APPROVED** [review]

**Comment by tjhei** [comment]
CI complains:
```error: file 'custom.css' not found or could not be opened```

File: doc/doxygen/CMakeLists.txt [hunk 1]
```diff
@@ -175,7 +175,8 @@ add_custom_command(
          ${CMAKE_CURRENT_BINARY_DIR}/footer.html
   COMMAND ${CMAKE_COMMAND} -E touch header.html
   COMMAND ${CMAKE_COMMAND} -E touch footer.html
-  COMMAND ${DOXYGEN_EXECUTABLE} -w html header.html footer.html style.css options.dox
+  COMMAND ${DOXYGEN_EXECUTABLE} -w html header.html footer.html
+                                   doxygen-awesome.css custom.css options.dox
```
Comment (gassmoeller) [inline doc/doxygen/CMakeLists.txt:179]: I think the problem with this PR is in this line. As far as I understand the doxygen documentation the .css files given here is one (and exactly one!) *output* file, not a list of css input files. In other words the fact that this file was called `style.css` before was not related to the `style.css` input file in doc/doxygen. I tested this locally for me, and with your change I get the same error message as the CI, but when I do the following everything works and builds with the new design:

```suggestion
                                   style.css options.dox
```
@pcafrica Do you want to give this a try and see what the testers say?

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Excited to see if this works! :-)

**Pull request merged**